lib/Runtime/Executor/ThreadPoolExecutor.h

```diff
@@ -25,6 +26,32 @@
 namespace glow {
 namespace runtime {
 
+/// This class implements a simple barrier with which to wait for all threads
```

Comment:
You've implemented a semaphore here. These kind of low level primitives are generally much more efficient when you use OS level primitives (e.g. mach semaphore, futex syscall). Not for now but important in the future.  I wonder if we should break it out into it's own class/file right away and make that work easier.